### PR TITLE
Migrate Supabase from v1 to v2

### PR DESCRIPTION
**Pull request by ezhil56x**
/claim #20
- Migrated Data methods
- Migrated Auth
- Realtime methods

**Still to do**
- Cookie method

### CODE REVIEW

File: web/hooks/useAuthRedirect.js [hunk 1]
```diff
@@ -6,22 +6,34 @@ const useAuthRedirect = () => {
 	const router = useRouter();
 
 	useEffect(() => {
-		const user = supabase.auth.user();
+		const checkAuthentication = async () => {
+			try {
+				const {
+					data: { session },
+				} = await supabase.auth.getSession();
+				const { user } = session;
```
[the remark this answers is not in the document]
Comment (ezhil56x) [reply web/hooks/useAuthRedirect.js:14]: Yes, missing session is correctly handled

**Comment by ezhil56x** [comment]
> @ezhil56x can you revert the formatting changes in the `.js` files? makes it hard to review the PR completely

How can I format only a particular file?
In some files, formatting has been changed on my commit

**Comment by rishi-raj-jain** [comment]
@ezhil56x 

You can squash your commits first, and then run prettier for the whole directory.

**Comment by ezhil56x** [comment]
> @ezhil56x 
> 
> You can squash your commits first, and then run prettier for the whole directory.

I tried it but it formatted nearly 8k lines. Thought that it would be even more difficult to review the PR

**Comment by rishi-raj-jain** [comment]
> > @ezhil56x
> > You can squash your commits first, and then run prettier for the whole directory.
> 
> I tried it but it formatted nearly 8k lines. Thought that it would be even more difficult to review the PR

Oh damn, does that the mean project itself right now is not prettier'ed?

**Comment by ezhil56x** [comment]
> > > @ezhil56x
> > > You can squash your commits first, and then run prettier for the whole directory.
> > 
> > I tried it but it formatted nearly 8k lines. Thought that it would be even more difficult to review the PR
> 
> Oh damn, does that the mean project itself right now is not prettier'ed?

I think so

**Comment by rishi-raj-jain** [comment]
Oh alright.

**Comment by ezhil56x** [comment]
@matthiaskern

Can I revert my commits, prettify and then commit the changes again?

Hope it becomes easier to review my PR

File: web/hooks/useAuthRedirect.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthiaskern) [reply web/hooks/useAuthRedirect.js]: `session` seems to be nullable according to the supabase types, can we cover this?

**Comment by matthiaskern** [comment]
@ezhil56x 🙇 didn't mean to cause this much extra work

**Comment by ezhil56x** [comment]
> @ezhil56x 🙇 didn't mean to cause this much extra work

No problem on it. Formatted the repo just before some time. Make my migration commits

File: web/hooks/useAuthRedirect.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ezhil56x) [reply web/hooks/useAuthRedirect.js:14]: > Does this handle an error or missing session correctly?

Just noticed this, will fix it and also will handle null in the next commit

[the remark this answers is not in the document]
Comment (ezhil56x) [reply web/hooks/useAuthRedirect.js]: @matthiaskern I made the changes. I hope it is correctly implemented

**Comment by ezhil56x** [comment]
> hey @ezhil56x, do you plan to finish this? if not, I can finish it and we can split up the bounty

Only some part is left out. Will contact you if I need some help

**Comment by ologbonowiwi** [comment]
sure, no problem! :rocket:

**Pull request closed without merging**